tools/cedict.py

```diff
@@ -0,0 +1,62 @@
+class Cedict:
+    def __init__(self, result: str, has_next_page: bool):
+        self.formatted_result = result
+        self.has_next_page = has_next_page
+
+
+with open("static/cedict_ts.u8") as origin_file:
+    cedict_data = origin_file.readlines()
+
+
+def get_cedict_result(query: str,
+                      required_page: int,
+                      entries_per_page: int) -> Cedict:
+    last_page = 0
```

Comment:
Agree about using **ceditct** as an attribute of **Cedict**, actually the reason i didn't do it at the time of opening the pull request is because i was lazy xd

as for **get_cedict_result()** - here it should be a part of a different class then, for example **CedictFetcher** class that would return **Cedict** object, to follow the Single Responsibility principle. Because **Cedict** is just a **data class**

Reply:
Agree about using ceditct as an attribute of Cedict, actually the reason i didn't do it at the time of opening the pull request is because i was lazy xd

as for get_cedict_result() - here it should be a part of a different class then, for example CedictFetcher class that would return Cedict object, to follow the Single Responsibility principle. Because Cedict is just a data class

Reply:
I would have only one class in `cedict.py` named `Cedict`, with `Cedict.__init__` for read and save dict data from file, `Cedict.get_result()` for getting results which return a list.

inside `command_cedict` plugin, I would just initialize `cedict = Cedict()` once, and use `result, has_next_page = cedict.get_result()` inside handlers for querying results.

Do you think this breaks Single Responsibility principle?

I think I got it why you use a dataclass here, list can have multiple items, and the code is too long for access dict keys, instead of returning a list or list, it better to return a dataclass.

you will probably get `too-few-public-methods (R0903)` from your `Cedict()` class if you check it by using pylint. I've never used a dataclass before, but the proper way to use it is probably using `dataclasses.dataclass`[1] (use `@dataclass` decorator and no `__init__`). I would like to have a separate class named `CedictResult` in this case.

1. https://realpython.com/python-data-classes/